src/types.rs

```diff
@@ -317,6 +318,176 @@ asn_integer_type! {
+            }
+
+            fn wrapping_add(self, other: Self) -> Self {
+                self.wrapping_add(other)
+            }
+        }
+
+        integer_type_decode!($($ts)*);
+    };
+    ((unsigned $t1:ty, $t2:ty), $($ts:tt)*) => {
+        impl IntegerType for $t1 {
+            const WIDTH: u32 = <$t1>::BITS;
+
+            fn try_from_bytes(
+                input: &[u8],
+                codec: crate::Codec,
+            ) -> Result<Self, crate::error::DecodeError> {
+                Self::try_from_unsigned_bytes(input, codec)
+            }
+
+            fn try_from_signed_bytes(
+                input: &[u8],
+                codec: crate::Codec,
+            ) -> Result<Self, crate::error::DecodeError> {
+                Ok(<$t2>::try_from_bytes(input, codec)? as $t1)
+            }
+
+            fn try_from_unsigned_bytes(
+                input: &[u8],
+                codec: crate::Codec,
+            ) -> Result<Self, crate::error::DecodeError> {
+                const BYTE_SIZE: usize = (<$t1>::BITS / 8) as usize;
+                if input.is_empty() {
+                    return Err(crate::error::DecodeError::unexpected_empty_input(codec));
+                }
+
+                let mut bytes = [0x00; BYTE_SIZE];
+                let start = bytes.len() - input.len();
+                bytes[start..].copy_from_slice(input);
+
+                Ok(Self::from_be_bytes(bytes))
+            }
+
+            fn wrapping_add(self, other: Self) -> Self {
+                self.wrapping_add(other)
+            }
+        }
+
+        integer_type_decode!($($ts)*);
+    };
+    (,) => {};
+    () => {};
+}
+
+integer_type_decode!(
+    (unsigned u8, i8),
+    (signed i8, u8),
+    (unsigned u16, i16),
+    (signed i16, u16),
+    (unsigned u32, i32),
+    (signed i32, u32),
+    (unsigned u64, i64),
+    (signed i64, u64),
+    (unsigned u128, i128),
+    (signed i128, u128),
+    (unsigned usize, isize),
+    (signed isize, usize),
+);
+
+impl IntegerType for Integer {
+    const WIDTH: u32 = u32::MAX;
+
+    fn try_from_bytes(
+        input: &[u8],
+        codec: crate::Codec,
+    ) -> Result<Self, crate::error::DecodeError> {
+        if input.is_empty() {
+            return Err(crate::error::DecodeError::unexpected_empty_input(codec));
+        }
+
+        Ok(Integer::from_signed_bytes_be(input))
+    }
+
+    fn try_from_signed_bytes(
+        input: &[u8],
+        codec: crate::Codec,
+    ) -> Result<Self, crate::error::DecodeError> {
+        Self::try_from_bytes(input, codec)
+    }
+
+    fn try_from_unsigned_bytes(
+        input: &[u8],
+        codec: crate::Codec,
+    ) -> Result<Self, crate::error::DecodeError> {
+        if input.is_empty() {
+            return Err(crate::error::DecodeError::unexpected_empty_input(codec));
+        }
+
+        Ok(BigUint::from_bytes_be(input).into())
+    }
+
+    fn wrapping_add(self, other: Self) -> Self {
+        self + other
+    }
```

Comment:
Why does this not use the actual wrapping_add method? On debug this will panic.

Reply:
that's the impl for `Integer`/`BigInt`, which doesn't have wrapping semantics because its not possible to overflow (and thus, there's no `BigInt::wrapping_add` method that exists)